### PR TITLE
Handle scenario of no darc updates for a given repo

### PR DESCRIPTION
**Pull request by MattGal**
Since this isn't necessarily indicative of anything wrong, just print warning and exit.  Also print output when we have git errors.

https://github.com/dotnet/core-eng/issues/11993

### CODE REVIEW

**Comment by MattGal** [comment]
https://dnceng.visualstudio.com/internal/_build/results?buildId=971547&view=results

File: eng/validation/build-arcadewithrepo.ps1 [hunk 1]
```diff
@@ -202,7 +204,15 @@ $barBuildId = ([regex]"\d+").Match($barBuildIdString).Value
...
 Set-Location $(Get-Repo-Location $global:githubRepoName)
-& $darc update-dependencies --id $barBuildId --github-pat $global:githubPAT --azdev-pat $global:azdoToken --password $global:bartoken
+
+$darcOutput = & $darc update-dependencies --id $barBuildId --github-pat $global:githubPAT --azdev-pat $global:azdoToken --password $global:bartoken
+Write-Output $darcOutput
+if ($darcOutput -eq "Found no dependencies to update.")
+{
+   ## Log a warning so this isn't hidden, but it's not necessarily a problem either, just exit.
+   Write-Host "##vso[task.logissue type=warning]Darc returned no dependencies to update for this repo"
+   exit 0
```
Comment (missymessa) [inline eng/validation/build-arcadewithrepo.ps1:214]: Let's have this return a failure. If the dependencies fail to update in this scenario, then we have an issue somewhere that needs to be fixed, and this particular pipeline will end up returning a false positive on a passing build with the latest Arcade version.

Comment (MattGal) [reply eng/validation/build-arcadewithrepo.ps1:214]: OK cool.  I'll promote it to error too.

Comment (MattGal) [reply eng/validation/build-arcadewithrepo.ps1]: Sure, also kicked off this version here; https://dnceng.visualstudio.com/internal/_build/results?buildId=971553&view=results

**Review by missymessa — APPROVED** [review]
LGTM shipit

**Pull request merged**